src/Core/Services/MetadataProviders/MetadataProviderFactory.cs

```diff
@@ -64,7 +64,8 @@ public void OnConfigChanged(object? sender, HotReloadEventArgs args)
         {
             _metadataProviders.Clear();
             ConfigureMetadataProviders();
-            InitializeAsync().Wait();
+            // Blocks the current thread until initialization is finished.
+            this.InitializeAsync().GetAwaiter().GetResult();
```

Comment:
Now that Abhishek's PR is in, can we use what he merged? 
`GetAwaiter().GetResult();` should only be used by the compiler. See: https://www.nikouusitalo.com/blog/why-is-getawaiter-getresult-bad-in-c/

Reply:
Wait() has the same problem according to that article except it also wraps any exceptions with its own exception handling.